### ISSUE TITLE
URI escaping in `def ext_url` causes ${filename} variable not to work

### ISSUE THREAD

**Issue by maxp-edcast** [googleapis/google-cloud-ruby#4047]
#### Environment details

  - Gem name and version: `google-cloud-storage` version 0.24.0

#### Steps to reproduce

  1. Call `post_object` with a key containing `${filename}`.
  2. Inspect `.fields[:key]` on the returned object.
  3. Note that `${filename}` has been changed to `$%7Bfilename%7D`
  4. Try and perform a presigned post in the browser, using the escaped key
  5. The file uploads, but the filename has not been replaced as it's supposed to. The uploaded file has the literal path `$%7Bfilename%7D`

#### Code example

```ruby
client = Google::Cloud::Storage.new project: "xxxxxxxxxx"

key = "${filename}"

post_object = client.bucket("some-bucket").post_object(key,  policy: {
  expiration: (Time.now + 1.hour).to_i,
  conditions: [
    ["starts-with", "$key", "" ],
    ["starts-with", "$Content-Type", "" ],
    { acl: 'public-read' },
    { success_action_status: '201' }
  ]
}

puts post_object.fields[:key]
# => $%7Bfilename%7D
```

#### NOTES

This seems to be caused by `def ext_path` in `lib/google/cloud/storage/file.rb`.

I can make the code work by manually unescaping the key after it's returned from `def post_object`. But this was very tricky to debug and I don't think this should be the expected behavior. 

`${filename}` is documented [here](https://cloud.google.com/storage/docs/xml-api/post-object) as a way to set the key of the uploaded file dynamically. However, the gem is URI escaping it and making it nonfunctional.

**Comment by quartzmo** [comment]
@maxp-edcast Thank you for opening this issue. I'll take a look at it this afternoon or this weekend.

**Comment by quartzmo** [comment]
@maxp-edcast Thank you for your patience, I should be able to look into this soon.

**Comment by quartzmo** [comment]
Original [Stack Overflow question](https://stackoverflow.com/questions/58001683/posting-object-to-google-cloud-storage-doesnt-replace-filename-variable)

**Comment by frankyn** [comment]
Hi @maxp-edcast,

could you tell more about why the filename should be a replaceable token `${filename}`? I would expect it to be the name that the object should be after upload is complete.

The current support is working as expected because `${}` are special characters and must be encoded.

**Comment by quartzmo** [comment]
Fix released in [google-cloud-storage v1.24.0](https://rubygems.org/gems/google-cloud-storage/versions/1.24.0).

**Comment by maxp-hover** [comment]
HI @frankyn I'm sorry I didn't respond to your question, I changed jobs and haven't logged into my maxp-edcast account since then (I should stop using job-specific github accounts, I guess). But, `${filename}` is used because in a presigned post we don't know the filename beforehand. For example in my browser I would see a file picker and I would select "cat.jpg", this would be the filename that is used. As I mentioned this is how it works with AWS S3. At the time I was working on an interface that would be able to build presigned posts for both providers, which I why I was requesting feature parity
